lib/trie/branch/branch.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package branch
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/trie/node"
+)
+
+var _ node.Node = (*Branch)(nil)
+
+// Branch is a branch in the trie.
+type Branch struct {
```

Comment:
did you intend for the stutter in this package? `branch.Branch`? I assumed that it would still be under the `trie` package.

Reply:
There is a lot of code for each node type, so splitting them is useful. I don't like the stuttering with names either, but I haven't figured out a solution to prevent that for now. Maybe in the future once there is more code re-used, I can move them to the lib/trie/node package instead.

Reply:
yea `node.Branch` and `node.Leaf` makes a lot of sense.  I don't mind having the interface and the implementations of that interface in the same pacakge.

Reply:
Done, see https://github.com/ChainSafe/gossamer/pull/2077#issuecomment-987742040